text/0000-promote-aarch64-unknown-linux-gnu-to-tier1.md

```diff
@@ -0,0 +1,188 @@
+
+ - The integration has been verified to produce green runs once patches from the two outstanding PRs are in place.
+
+**2.a. The long term viability of the existence of a target specific ecosystem should be clear.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - That said, Arm AArch64 silicon is either already prevalent or is en-route to prevalance in a wide spectrum of application domains ranging from 'traditional' embedded systems at one end of the spectrum, on to mobile phones, clam-shell devices, desktops, vehicle autonomy controllers, datacenter servers etc all the way to high performance super-computers.
+
+ - The evidence to that effect is too numerous to quote but generally easy to verify openly. 
+
+ - It is fair to state that this is an ongoing reality which is unlikely to stop trending upwards and sidewards for the forseeable future.
+
+ - Software stacks built for those domains predominantly use an AArch64 Linux kernel build.
+
+ - Rust presents an attractive value proposition across all such domains, irrespective of the underlying processor architecture.
+
+ - **As such, the Rust aarch64-unknown-linux-gnu target's ecosystem presents very strong viability for the long term.**
+
+**2.b. The long term viability of supporting the target should be clear.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - It is worth calling out, in the same vein as the previous point, that given the increasing prevalance of AArch64 silicon deployments and given Rust's general value proposition, **supporting the Rust aarch64-unknown-linux-gnu target presents very strong viability for the long term.**
+
+**2.c. The target must have substantial and widespread interest within the Rust developer community.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - It is generally fair to state that **there is already substantial and widespread interest for the aarch64-unknown-linux-gnu target in the Rust developer commmunity**.
+
+ - It is also generally fair to state that there is a clear upward trend in the use of AArch64 systems as self hosted development environments. 
+
+ - Most major operating system environments support hosted development on AArch64 based systems and this trend is increasing.
+
+ - As a somewhat related note: Slow but steady progress is being made to support Windows AArch64 targets, initially for cross-platform development. This shall inevitably trend towards hosted development.
+
+ - As such, **it is very likely that developer interest in Rust on aarch64-unknown-linux-gnu will continue to increase in the medium to long term.**
+
+**2.d. The target must serve the interests of multiple production users of Rust across multiple organizations or projects.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - Most major Arm software ecosystem partners are either already using Rust extensively, or are building up to extensive use. A few publically known examples are Microsoft and Google. There are many more.
```

Comment:
I was surprised to see Google and Microsoft listed here, but not Amazon/AWS. Google and Microsoft both work on a lot of arm and aarch64 platforms, but predominantly Android, Fuchsia, Windows, and embedded systems, not aarch64-unknown-linux-gnu, and both are dipping their toes in with Rust. There's a substantial portions of Fuchsia written in Rust, but I don't know if any products are depending on Fuchsia yet. Microsoft has mentioned it an released some WinRT bindings and a research language inspired by Rust, but again, that seems like 

Amazon uses Rust for [Firecracker](https://firecracker-microvm.github.io/), a key portion of AWS Lambda and Fargate, they offer their [Graviton](https://aws.amazon.com/ec2/graviton/) servers which are aarch64-unknown-linux-gnu, and they mention using Rust and ARM extensively in their Nitro system. It seems to me that in terms of publicly known examples, Amazon relies more on Rust on aarch64-unknown-linux-gnu than either Google or Microsoft.

I'm not affiliated with Amazon, just felt like they are a better example here.

Reply:
This is somewhat paranthetical, but neither lambda nor fargate actually runs on Gravitron2 CPUs AFAIK.

Reply:
This is true, they use Rust in prod on x86_64 in the form of lambda and fargate, and they use aarch64-unknown-linux-gnu in prod in the form of their Graviton-based instances, and they use Rust on some kind of Arm in prod for their Nitro system (exact details of the platform aren't public, but they specifically mention that it uses Rust on Arm).

They also release [Firecracker builds for aarch64-unknown-linux-gnu](https://github.com/firecracker-microvm/firecracker/releases/tag/v0.21.1), they have ECS support on Graviton systems, and preview EKS support on Graviton systems, but they don't actually have their fully managed Lambda or Fargate on Graviton systems yet.

So, that's fair, the actual Lambda and Fargate products don't run on aarch64 at the moment, but they are clearly basing real products on both Rust and aarch64, and are producing builds of their open source Rust components for aarch64.

Anyhow, point is, they are fairly heavily committed to Rust and aarch64-unknown-linux-gnu, and are actively building software that the depend on in the open using Rust, and which targets aarch64-unknown-linux-gnu.

Though I suppose to be fair, Firecracker was originally a fork of crosvm, a VMM written in Rust for ChromeOS, which also runs on aarch64 systems which are shipping products, so Google does have some shipping products here too which are also public.

And of course, Google has a huge investment in aarch64-linux-android and aarch64-fuchsia, and Fuchsia uses Rust fairly heavily, while there are traces of Rust in Android but I'm not sure if it's in anything shipping yet.

Reply:
Hi again. 

Fair point about Amazon. Will fix. 

I simply mentioned orgs I was directly involved with on impulse. 

Thanks

Reply:
I've added a commit to address @lambda 's comments.

Also responded hopefully suitably to the other comments here.

I'm going to resolve this conversation. Happy to engage further as needed.